management-interface.go

```diff
@@ -118,23 +118,29 @@ func getDeviceName() string {
 
 // Return the serial number for the Raspberr Pi in the device.
 func getRaspberryPiSerialNumber() string {
-	if runtime.GOOS != "windows" {
-		// 'Nix.  Run /proc/cpuinfo command to get the info we want.
-		out, err := exec.Command("cat", "/proc/cpuinfo").Output()
-		if err != nil || len(out) == 0 {
-			return ""
-		}
-		// Extract the serial number. It's in the last row bar 1.  The last row of the output is blank.
-		rows := strings.Split(string(out), "\n")
-		lastRow := rows[len(rows)-2]
-		// Make sure the last row does in fact contain a serial number.
-		if !strings.Contains(strings.ToUpper(lastRow), "SERIAL") {
-			return ""
+
+	if runtime.GOOS == "windows" {
+		return ""
+	}
+
+	// Run /proc/cpuinfo command to get the info we want.
+	out, err := exec.Command("cat", "/proc/cpuinfo").Output()
+	if err != nil || len(out) == 0 {
+		return ""
+	}
+	
+	// Extract the serial number.
+	serialNumber := ""
+	rows := strings.Split(string(out), "\n")
+	for _, row := range rows {
+		if strings.Contains(strings.ToUpper(row), "SERIAL") {
+			serial := strings.Split(row, ":")
+			serialNumber = serial[1][1:]
```

Comment:
Instead of assuming there's exactly one space after the `:`, just call `strings.Trim` on `serialNumber`. This will remove any whitespace from either side.

Reply:
Also better check that there are 2 parts returned from Split. If there's not then keep going. Just in case some other line ends up with "serial" in it. In fact the most robust approach would be doing this for each row:

```go
parts := strings.Split(row, ":")
if len(parts) == 2 {
    field := strings.ToUpper(strings.Trim(parts[0]))
    if field == "SERIAL" {
       return strings.Trim(parts[1])
    }
}
```

Then you'll only match on the actual "Serial" line and not some other line that happen to have "serial" in it somewhere.